### PR TITLE
5 New Drink Recipes for Bartender

### PR DESCRIPTION
**Pull request by t-toasted**
## About the PR 
Adds in 5 new recipes for the bartender: iced coconut milk espresso, iced pineapple matcha, thai iced coffee, peppermint hot chocolate, and mexican hot chocolate.  

RECIPES:
Iced Coconut Milk Espresso: required_reagents = list("espresso" = 1, "ice" = 3, "coconut_milk" =2, "sugar" = 2)
Iced Pineapple Matcha: required_reagents = list("matcha" = 1, "juice_pineapple" = 1, "coconut_milk" = 2, "ice" = 1)
Thai Iced Coffee: required_reagents = list("coffee" = 3, "sugar" = 1, "milk" = 1, "ice" = 1)
Peppermint Hot Chocolate: required_reagents = list("chocolate" = 2, "mint" = 1, "milk" = 1)
Mexican Hot Chocolate: required_reagents = list("chocolate" = 2, "capsaicin" = 1, "milk" = 1)

## Why's this needed? 
Some bartenders like their mixtures to have actual names. Gives bartenders more ideas of things to serve up, for a barista feel, and encourages them to use more stuff from botany for their drinks.

## Changelog

```changelog
(u)Roddy
(*)Added in five new drink recipes for the bartender; Iced Coconut Milk Espresso, Iced Pineapple Matcha, Thai Iced Coffee, Peppermint Hot Chocolate, and Mexican Hot Chocolate!
```

### CODE REVIEW

**Comment by flrsh** [comment]
Could you make all the iced drinks an iced subtype just for the sake of organization?

**Comment by t-toasted** [comment]
> Could you make all the iced drinks an iced subtype just for the sake of organization?

Can do!

**Comment by t-toasted** [comment]
!merge upstream

File: code/modules/chemistry/Chemistry-Recipes.dm [hunk 1]
```diff
@@ -1844,6 +1844,55 @@ datum
 					holder.del_reagent("blood")
 					holder.del_reagent("bloodc")
 
+		iced_coconutmilkespresso
```
Comment (flrsh) [inline code/modules/chemistry/Chemistry-Recipes.dm:1847]:
```suggestion
		iced/coconutmilkespresso
```

File: code/modules/chemistry/Chemistry-Recipes.dm [hunk 2]
```diff
@@ -1844,6 +1844,55 @@ datum
...
+			name = "Iced Coconut Milk Espresso"
+			id = "icedcoconutmilkespresso"
+			result = "icedcoconutmilkespresso"
+			required_reagents = list("espresso" = 1, "ice" = 3, "coconut_milk" =2, "sugar" = 2)
+			result_amount = 8
+			mix_phrase = "The ice clinks against the container as you blend everything together."
+			mix_sound = 'sound/misc/drinkfizz.ogg'
+
+		iced_pineapplematcha
```
Comment (flrsh) [inline code/modules/chemistry/Chemistry-Recipes.dm:1856]:
```suggestion
		iced/pineapplematcha
```

File: code/modules/chemistry/Chemistry-Recipes.dm [hunk 3]
```diff
@@ -1844,6 +1844,55 @@ datum
...
+			name = "Iced Pineapple Matcha"
+			id = "icedpineapplematcha"
+			result = "icedpineapplematcha"
+			required_reagents = list("matcha" = 1, "juice_pineapple" = 1, "coconut_milk" = 2, "ice" = 1)
+			result_amount = 5
+			mix_phrase = "The milk mixes with the matcha in a soothing green."
+			mix_sound = 'sound/misc/drinkfizz.ogg'
+
+		iced_thaiicedcoffee
```
Comment (flrsh) [inline code/modules/chemistry/Chemistry-Recipes.dm:1865]:
```suggestion
		iced/thaiicedcoffee
```

File: code/modules/chemistry/Chemistry-Recipes.dm [hunk 4]
```diff
@@ -1844,6 +1844,55 @@ datum
...
+			name = "Thai Iced Coffee"
+			id = "thaiicedcoffee"
+			result = "thaiicedcoffee"
+			required_reagents = list("coffee" = 3, "sugar" = 1, "milk" = 1, "ice" = 1)
+			result_amount = 6
+			mix_phrase = "Everything mixes together nicely, releasing a sweet smell."
+			mix_sound = 'sound/misc/drinkfizz.ogg'
+
+		iced_thaiicedcoffee/thaiicedcoffee2
```
Comment (flrsh) [inline code/modules/chemistry/Chemistry-Recipes.dm:1874]:
```suggestion
		iced/thaiicedcoffee/thaiicedcoffee2
```

File: code/modules/chemistry/Reagents-FoodDrink.dm [hunk 5]
```diff
@@ -4411,6 +4411,66 @@ datum
 				flush(M, 3 * mult, flushed_reagents)
 				..()
 
+		fooddrink/icedcoconutmilkespresso
```
Comment (flrsh) [inline code/modules/chemistry/Reagents-FoodDrink.dm:4414]:
```suggestion
		fooddrink/iced/coconutmilkespresso
```

File: code/modules/chemistry/Reagents-FoodDrink.dm [hunk 6]
```diff
@@ -4411,6 +4411,66 @@ datum
...
+			fluid_r = 177
+			fluid_g = 143
+			fluid_b = 106
+			transparency = 200
+			taste = list("sweet", "cold")
+			description = "A creamy iced espresso, mixed with coconut milk."
+			reagent_state = LIQUID
+			thirst_value = 0.8
+
+		fooddrink/icedpineapplematcha
```
Comment (flrsh) [inline code/modules/chemistry/Reagents-FoodDrink.dm:4426]:
```suggestion
		fooddrink/iced/pineapplematcha
```

File: code/modules/chemistry/Reagents-FoodDrink.dm [hunk 7]
```diff
@@ -4411,6 +4411,66 @@ datum
...
+			fluid_r = 152
+			fluid_g = 184
+			fluid_b = 96
+			transparency = 200
+			taste = list("earthy")
+			description = "Tangy, yet refreshingly earthy."
+			reagent_state = LIQUID
+			thirst_value = 0.8
+
+		fooddrink/thaiicedcoffee
```
Comment (flrsh) [inline code/modules/chemistry/Reagents-FoodDrink.dm:4438]:
```suggestion
		fooddrink/iced/thaicoffee
```

File: code/modules/chemistry/Reagents-FoodDrink.dm [hunk 8]
```diff
@@ -4411,6 +4411,66 @@ datum
...
+			name = "thai iced coffee"
```
Comment (flrsh) [inline code/modules/chemistry/Reagents-FoodDrink.dm:4439]:
```suggestion
			name = "Thai iced coffee"
```

File: code/modules/chemistry/Reagents-FoodDrink.dm [hunk 9]
```diff
@@ -4411,6 +4411,66 @@ datum
...
+			fluid_g = 94
+			fluid_b = 43
+			transparency = 200
+			taste = list("minty", "chocolatey")
+			description = "Minty, creamy, and chocolatey; delicious!"
+			reagent_state = LIQUID
+			thirst_value = 0.8
+
+		fooddrink/mexicanhotchocolate
+			name = "mexican hot chocolate"
```
Comment (flrsh) [inline code/modules/chemistry/Reagents-FoodDrink.dm:4463]:
```suggestion
			name = "Mexican hot chocolate"
```

**Review by flrsh — CHANGES_REQUESTED** [review]

**Comment by wolfolotl** [comment]
thing I realized looking at this: for the hot chocolate recipes you'll want to add in duplicate formulas that use the chocolate milk reagent instead of chocolate and milk separately, or that reaction might occur first and lock the bartender out of the drink. similar things have happened with recipes that require tea and sugar but don't have a recipe with the sweet tea reagent itself

**Comment by t-toasted** [comment]
> thing I realized looking at this: for the hot chocolate recipes you'll want to add in duplicate formulas that use the chocolate milk reagent instead of chocolate and milk separately, or that reaction might occur first and lock the bartender out of the drink. similar things have happened with recipes that require tea and sugar but don't have a recipe with the sweet tea reagent itself

Oh! How didn’t I catch that. Yes yes, good idea, let me fix that

**Review by flrsh — APPROVED** [review]
Looks good to me. Could also make Mexican hot chocolate and peppermint hot chocolate subtypes of hot chocolate, but since there are only two kinds (right now) it's up to you.

**Comment by t-toasted** [comment]
> Looks good to me. Could also make Mexican hot chocolate and peppermint hot chocolate subtypes of hot chocolate, but since there are only two kinds (right now) it's up to you.

If I add more in the future, I’ll change them to be a subtype (now that I know how to too), but for now I’ll leave them be

**Pull request merged**